src/tini.c

```diff
@@ -133,8 +133,19 @@ int spawn(const signal_configuration_t* const sigconf_ptr, char* const argv[], i
 		}
 
 		execvp(argv[0], argv);
-		PRINT_FATAL("Executing child process '%s' failed: '%s'", argv[0], strerror(errno));
-		return 1;
+	    	// execvp will only return on an error so make sure that we check the errno
+		// and exit with the correct return status for the error that we encountered
+		int status = EXIT_FAILURE;
+		switch errno {
+		case ENOENT:
+			status = 127;
+			break;
+		case EACCES:
+			status = 126;
+			break;
+		}
+		fprintf(stderr, "%s\n", strerror(errno));
```

Comment:
Any objections to using the PRINT_X macros here?

The current message is a little awkward, so we can certainly make it more succinct (especially since the concept of the "child process" is probably irrelevant to end-users here), e.g.: `Execute %s failed: %s`.

I'm also totally fine with conditionally making those macros a little less verbose / more anonymous (e.g. remove the severity levels and the tini prefix) when `TINI_NO_ARGS` is set  if that's your concern though (maybe we should rename this `TINI_DOCKERINIT` though in this case..! 😄).

Just let me know!